pkg/cmd/build/build.go

```diff
@@ -175,41 +174,14 @@ func (ob *OktetoBuilder) buildWithOkteto(ctx context.Context, buildOptions *type
 
 	buildkitWaiter := buildkit.NewBuildkitClientWaiter(buildkitClientFactory, ioCtrl)
 
-	err = buildkitWaiter.WaitUntilIsUp(ctx)
-	ob.metadata.WaitForBuildkitAvailableTime = buildkitWaiter.GetWaitingTime()
-	if err != nil {
-		return err
-	}
+	reg := registry.NewOktetoRegistry(GetRegistryConfigFromOktetoConfig(ob.OktetoContext))
+	buildSolver := buildkit.NewBuildkitRunner(buildkitClientFactory, buildkitWaiter, reg, run, ioCtrl)
 
-	buildkitClient, err := buildkitClientFactory.GetBuildkitClient(ctx)
-	if err != nil {
+	if err := buildSolver.Run(ctx, opt, buildOptions.OutputMode); err != nil {
 		return err
 	}
 
-	err = run(ctx, buildkitClient, opt, buildOptions.OutputMode, ioCtrl)
-	if err != nil {
-		if shouldRetryBuild(err, buildOptions.Tag, ob.OktetoContext) {
-			ioCtrl.Logger().Infof("Failed to build image: %s", err.Error())
-			ioCtrl.Logger().Infof("isRetry: %t", ob.isRetry)
-			if !ob.isRetry {
-				retryBuilder := NewOktetoBuilder(ob.OktetoContext, ob.Fs)
-				retryBuilder.isRetry = true
-				err = retryBuilder.buildWithOkteto(ctx, buildOptions, ioCtrl, run)
-			}
-		}
-		err = getErrorMessage(err, buildOptions.Tag)
-		return err
-	}
-
-	var tag string
-	if buildOptions != nil {
-		tag = buildOptions.Tag
-		if buildOptions.Manifest != nil && buildOptions.Manifest.Deploy != nil {
-			tag = buildOptions.Manifest.Deploy.Image
-		}
-	}
```

Comment:
Where is this logic now? Isn't it needed?

Reply:
I checked that with deploy remote we were already capturing the correct error. We changed from checking if the buildOptions was empty infer that is a deploy/destroy/test operation to capture the error directly from the error itself (changed in 3.0)